rfcs/20201202-dnnl-verbose-extension/README.md

```diff
@@ -0,0 +1,40 @@
+# RFC: DNNL_VERBOSE functionality extension
+
+## Motivation
+
+oneDNN users wish to profile the library calls via chrome trace, which is
+popular among different products. In order to have it working properly,
+DNNL_VERBOSE should dump a timestamp of each call, otherwise it's impossible
+to pin a call to a certain point in the timeline. It may be either an absolute
+time value in milliseconds or relative to a certain start point dumped once.
+Both ways will let to profile each call through the whole program execution.
+
+## Proposal
+
+To keep the current behavior intact, the proposal is to introduce a modifier
+which will be enabled by additional environment variable
+`DNNL_VERBOSE_TIMESTAMP`. When set to `0` (the default), it does not affect the
+existing output of DNNL_VERBOSE. When set to `1`, it adds absolute time in
+milliseconds. There are two options where to put this timestamp:
+* In front of the line. This follows the standard logging practices but will
+  break existing parsing solutions (preferred).
+* In the end of the line. This will preserve the existing solutions to work
+  whether the variable is enabled or not, but may bring some confusion which
+  time is actual performance, but which is a timestamp. Besides, to utilize the
+  new ability, adjustments to script should be done any way.
+
+## Example
+
+> 1.60704e+12:dnnl_verbose,exec,cpu,reorder,rnn_data_reorder,undef,src_f32::blocked:abcd:f0 dst_u8::blocked:abcd:f0,,,2x3x224x224,3.3728
```

Comment:
Another option is to put the timestamp after `dnnl_verbose` : `dnnl_verbose,<timestamp>,(exec|create),...`. I realized it might better fit the existing verbose format.

`1.60704e+12` - probably better to use an integer.

Reply:
I've updated the format. It will use whatever `std::to_string` will format it into.

I've also added an option with timestamp being a second argument and mark this option as preferred. Please review.

Thanks.